cider/exec/nextgen/NextgenModule.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef EXEC_NEXTGEN_MODULE_H
+#define EXEC_NEXTGEN_MODULE_H
+
+#include <algorithm>
+#include <memory>
+#include "exec/nextgen/Context.h"
+#include "exec/nextgen/parsers/Parser.h"
+#include "exec/nextgen/transformer/Transformer.h"
+#include "exec/template/RelAlgExecutionUnit.h"
+#include "jitlib/llvmjit/LLVMJITModule.h"
+
+namespace cider::exec::nextgen {
+
+static std::unique_ptr<Context> compile(const RelAlgExecutionUnit& ra_exe_unit,
```

Comment:
should be substrait? Like https://github.com/intel/BDTK/pull/203/files#diff-3bfcd83833a1bb8c7af1724cfef238b8f17379311a36d21ff55d84c519dcb0c1R45

Reply:
it is called by compile method in CiderCompileModule